src/hooks.ts

```diff
@@ -2,34 +2,111 @@ import { useState, useEffect } from 'react';
 import { State, Resource } from 'ketting';
 
 type UseResourceResult<T> = {
+  /**
+   * 'true' if resource has not yet been fetched from the server.
+   */
   loading: boolean
+
+  /**
+   * Contains an Error object when an operation has failed.
+   */
   error: null | Error,
+
+  /**
+   * Data from the resource
+   */
   data: T,
+
+  /**
+   * Updates the local data cache and dispatch change events.
+   */
+  updateData: (data: T) => void,
```

Comment:
Could this be `updateCache`?

Reply:
I went with `updateData` because it refers to the `data` property, just like `updateResourceState` refers to updating `resourceState`.

I see now that most react hooks examples use pairs of `data / setData` and not `data / updateData`, so maybe this is worth fixing for consistency here.

I guess the intent here is a very advanced `useState`

Reply:
i think `setData` would work very nicely with this